src/main/java/org/pegdown/Parser.java

```diff
@@ -557,13 +535,18 @@ Node parseListBlock(StringBuilderVar block) {
         Node innerRoot = parseInternal(block);
         setContext(context); // we need to save and restore the context since we might be recursing
         block.clearContents();
+        //debugMsg("parsed list block " + innerRoot.toString() + " adjusting indices by " + getContext().getValueStack().peek(), block.getString());
         return withIndicesShifted(innerRoot, (Integer) context.getValueStack().pop());
     }
     
     Node withIndicesShifted(Node node, int delta) {
-        ((AbstractNode) node).shiftIndices(delta);
-        for (Node subNode : node.getChildren()) {
-            withIndicesShifted(subNode, delta);
+        if (delta != 0)
+        {
```

Comment:
Not on purpose. I usually do a reformat on the file but in this case was afraid that it would make seeing the delta impossible. So I refrain from reformatting. Not used to keep track of format manually. IntelliJ has me spoiled.

Best regards,


Vladimir.

> On Aug 22, 2015, at 5:33 PM, Mathias <notifications@github.com> wrote:
> 
> In src/main/java/org/pegdown/Parser.java <https://github.com/sirthias/pegdown/pull/181#discussion_r37700003>:
> 
> >          return withIndicesShifted(innerRoot, (Integer) context.getValueStack().pop());
> >      }
> >
> >      Node withIndicesShifted(Node node, int delta) {
> > -        ((AbstractNode) node).shiftIndices(delta);
> > -        for (Node subNode : node.getChildren()) {
> > -            withIndicesShifted(subNode, delta);
> > +        if (delta != 0)
> > +        {
> Why do you switch to another indentation style here?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/sirthias/pegdown/pull/181/files#r37700003>.
>